internal/generated_file_test/bin.js

```diff
@@ -0,0 +1,48 @@
+const fs = require('fs');
+const path = require('path');
+const runfiles = require(process.env['BAZEL_NODE_RUNFILES_HELPER']);
+
+function main(args) {
+  const [mode, golden_no_debug, golden_debug, actual] = args;
+  const actualPath = runfiles.resolveWorkspaceRelative(actual);
+  const debugBuild = /\/bazel-out\/[^/\s]*-dbg\//.test(actualPath);
+  const golden = debugBuild ? golden_debug : golden_no_debug;
+  const actualContents = fs.readFileSync(actualPath, 'utf-8').replace(/\r\n/g, '\n');
+  const goldenContents =
+      fs.readFileSync(runfiles.resolveWorkspaceRelative(golden), 'utf-8').replace(/\r\n/g, '\n');
+
+  if (actualContents === goldenContents) {
+    return 0;
+  }
+  if (mode === '--out') {
+    // Write to golden file
+    fs.writeFileSync(runfiles.resolveWorkspaceRelative(golden), actualContents);
+    console.error(`Replaced ${path.join(process.cwd(), golden)}`);
+    return 0;
+  }
+  if (mode === '--verify') {
+    const unidiff = require('unidiff');
```

Comment:
Nothing needs to be done to ensure `unidiff` is available?

I know the `@bazel/*` npm packages use `peerDependencies`, but how does it work for rules built directly into rules_nodejs?

Reply:
I see `@npm//unidiff` is passed into the `nodejs_test|binary(data)`, but is that the consumers `@npm//` and we assume the package exists?

Reply:
Immediate answer is this needs some e2e test to show how it's used outside our workspace.
Yes I think that will show it's broken like you say. This rule has to be in an npm package or else unidiff has to be vendored.

Reply:
That could always be a followup, wouldn't change what you've done.